funnel/models/event.py

```diff
@@ -125,6 +126,24 @@ class Participant(BaseMixin, db.Model):
     __table_args__ = (db.UniqueConstraint('proposal_space_id', 'email'),)
 
     @classmethod
+    def get(cls, space, email, fields={}, create=False):
+        participant = cls.query.filter_by(proposal_space=space, email=email).first()
+        if create and not participant:
+            participant = cls(
+                fullname=fields.get('fullname'),
+                email=email,
+                phone=fields.get('phone'),
+                twitter=fields.get('twitter'),
+                job_title=fields.get('job_title'),
+                company=fields.get('company'),
+                city=fields.get('city'),
+                events=fields.get('events', []),
```

Comment:
IMO, all setters should be moved out of the `get` method. Return an empty participant object and let the caller set/update the fields.